### PR TITLE
Ria 6424 appeal submission summary

### PR DESCRIPTION
**Pull request by hussain-faruk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6424
### Change description ###
Added some defs to AsylumCaseDefinitions to extract asylum case data
Added some putters to place case data in field values which docmosis can use in order to generate the appeal submission summary

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by hussain-faruk** [comment]
After discussing with Neha and Marcello, extracting noms number is proving quite difficult using AsylumCaseDefinition, as the field is coming through as some complect data type rather than a simple string. So I have manually extracted the data and used regex matching to get the data I want. This is not an optimal solution but I cannot see how else it can be done. Nowhere else in the entire codebase (using ctrl+n) is noms number being extracted from asylumcasedata, so I cannot go by others work.

Same applied to prisoner release date.

**Comment by hussain-faruk** [comment]
In a couple of tests, when asserting if number of fields in case data == number of fields in the mapped templateFieldValues, the values have been hard coded. I have not corrected this however I believe this should be ,changed.

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplate.java [hunk 1]
```diff
@@ -64,6 +64,76 @@ public Map<String, Object> mapFieldValues(
...
+            if (isAcceleratedDetainedAppeal.equals(YesOrNo.YES)) {
+                sb.append(" - Accelerated");
+                fieldValues.put("isAcceleratedDetainedAppeal", YesOrNo.YES);
+            }
+            fieldValues.put("detentionStatus", sb.toString());
+            sb.setLength(0);
+
+            String detentionFacility = asylumCase.read(DETENTION_FACILITY, String.class).orElse(null);
+            String detentionFacilityName = "";
+            switch (detentionFacility) {
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplate.java:81]: Just a suggestion, can you take the block related to setting fields of detention facility and move to a private method?

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplateTest.java [hunk 2]
```diff
@@ -54,7 +54,20 @@ public class AppealSubmissionTemplateTest {
...
     private String appellantAddressPostCode = "AB1 2CD";
     private String appellantAddressCountry = "Iceland";
+    private YesOrNo appellantInDetention = YesOrNo.YES;
+    private String detentionFacilityPrison = "prison";
+    private String detentionFacilityIrc = "immigrationRemovalCentre";
+    private String detentionFacilityOther = "other";
+    private String prisonName = "HMP Gartree";
+    private String nomsNumber = "{noms=Noms1234}";
+    private String ircName = "MoJ IRC";
+    private String otherName = "{otherName=Other MoJ Facility}";
```
[the remark this answers is not in the document]
Comment (hussain-faruk) [reply src/test/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/AppealSubmissionTemplateTest.java:66]: No. These fields come through from the UI to case asylum in the same format, so in the tests I have to replicate the same format. This is why the formatComplexString() method exists in AppealSubmissionTemplate.

**Review by neha-aggarw — APPROVED** [review]

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Pull request merged**